Userland/Libraries/LibCore/MachPort.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Platform.h>
+
+#ifndef AK_OS_MACH
+#    error "MachPort is only available on Mach platforms"
+#endif
+
+#include <AK/Error.h>
+#include <AK/Noncopyable.h>
+#include <mach/mach.h>
+
+namespace Core {
+
+class MachPort {
```

Comment:
Might be helpful for future devs if we throw some upstream links on this class? Maybe just googling "apple mach port" would bring me to the right thing, but would be nice to be sure